### ISSUE TITLE
Add ability to block/unblock users

### ISSUE THREAD

**Issue by CouchersBot** [Couchers-org/couchers#4165]
Add the ability to block/unblock users. This is necessary for user safety.

The backend is already done for this.The API is here: 
https://github.com/Couchers-org/couchers/blob/develop/app/proto/blocking.proto

**BLOCKING FLOW:**
- On a user's profile page, user clicks on teal flag icon:
<img width="321" alt="Image" src="https://github.com/user-attachments/assets/d3259a1b-a32d-43da-b68c-211a284008e7" />

- Add text "Report or Block" to the right of the flag icon

-  A modal pops up with this design:

<img width="517" alt="Image" src="https://github.com/user-attachments/assets/d444870b-69dc-4ae0-ba08-5e9f676f2af0" />

- The report checkbox should be activated by default, and the block user checkbox should be deactivated by default.

- If a user deactivates the report checkbox, the dropdown menu and input box collapse/disappear and it looks like this:

<img width="645" alt="Image" src="https://github.com/user-attachments/assets/01355ee8-97a8-4a9d-aa6d-5758ce8a5853" />

- Upon submitting, a basic confirmation dialog must be shown to prevent accidental reports or blocks.

- If the user unchecks both boxes, then the "Submit" button should become inactive (greyed out) since there is nothing to submit in that case. Alternatively, we could use an error alert to warn the user if they try to click the button when neither checkbox is activated. In this case, use this error alert text:
`To submit, at least one checkbox must be selected`

Notes:
If the API supports it, we could display "Block [name]" instead of the generic "Block this person".

**UNBLOCKING FLOW:**
-  Add to "my connections" page (https://couchers.org/connections). 
- The blocked users should appear in the bottom-right quadrant of the matrix (next to "friend requests you sent" and below "your friends"). 
- On mobile it should appear at the very bottom of the page. 

========================================
Subject: Blocking
Description:
Used website

Results:
It's essential to be able to block some members e.g. who may have been a previous bad experience on Couchsurfing...

Backend version: develop-d04e9dbe
Frontend version: develop-d04e9dbe
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:78.0) Gecko/20100101 Firefox/78.0
Page: https://app.couchers.org/
User: smersh (1396)

**Comment by darrenvong** [comment]
Are we leaving this for support or should this actually be implemented in the frontend? Looks like there are some APIs available in `blocking.proto`

**Comment by lucaslcode** [comment]
Yes we should

**Comment by jesseallhands** [comment]
Why: User blocking should be implemented to address issues mentioned in my comment on #3526 related to requests and to general unwanted messages/spam. This feature has been requested multiple times already per support tickets.

I believe the block user API already exists, so this should just require an update of the current report modal.

#### Current design:
Upon clicking on the teal flag icon either below the profile picture on the profile page or on the messaging page the report modal opens:
![image](https://github.com/user-attachments/assets/4cd264be-aa52-4e1c-9c6b-4a07bf75a8b1)

#### New design:
![image](https://github.com/user-attachments/assets/0098af0c-c65f-4b8d-8f96-9f6267dc4f44)

The report checkbox should be activated by default, and the block user checkbox should be deactivated by default.

If a user deactivates the report checkbox, the dropdown menu and input box collapse/disappear and it looks like this:
![image](https://github.com/user-attachments/assets/96a9cfd1-c176-4997-a663-4849149923dd)

Upon submitting, a basic confirmation dialog must be shown to prevent accidental reports or blocks.

#### Notes:
- If the API supports it, we could display "Block [name]" instead of the generic "Block this person". 
- Initially, unblocking will be a manual process done by emailing support, but eventually we can update the "my connections" page so that users can manage blocked users from there.

#### Steps:

Backend (should be done?)
Design (done)
Frontend
Testing
Help Pages (in progress)

**Comment by nabramow** [comment]
Do we have the backend done to filter out blocked users from user searches, comments in groups, etc.? @aapeliv @jesseallhands 

Generally once you block someone you don't see their posts, comments, etc. anymore nor see them in search results.

If we don't have that, let's make a separate ticket to do first?


Also, should we not do the unblocking in this feature too? Doesn't seem too crazy, just need to decide where to put it, then can just make a list with user avatar, name and a button that says "Unblock".

**Comment by jesseallhands** [comment]
Backend should be done, but @aapeliv can confirm.

If it's not too crazy, then I'd say yes, we should include the unblocking in this feature too. This can be done on the "my connections" page (https://couchers.org/connections). The blocked users should appear in the bottom-right quadrant of the matrix (next to "friend requests you sent" and below "your friends"). On mobile it should appear at the very bottom of the page. At some point we should improve navigation of that page on mobile because it's a lot of scrolling if you have a lot of connections.

**Comment by jesseallhands** [comment]
I forgot to mention: If the user unchecks both boxes, then the "Submit" button should become inactive (greyed out) since there is nothing to submit in that case. Alternatively, if the boxes are both unchecked and the user clicks submit, we could use an error alert to warn the user. In this case, use this error alert text: 

`To submit, at least one checkbox must be selected`

**Comment by nabramow** [comment]
Hey @sijav are you still planning on working on this?
